src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -27,25 +31,65 @@ namespace CoreNodeModels.Input
     [OutPortDescriptions("Selected Color.")]
     public class ColorPalette : NodeModel
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private DSColor prevColor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
```

Comment:
to maybe clarify a bit more - we don't want undo's and redo's to alter the undo stack.